Library/Homebrew/formula.rb

```diff
@@ -2185,16 +2180,32 @@ def to_hash
       }
     end
 
+    # TODO: can we implement these in Formulary?
+    if self.class.loaded_from_api && Homebrew::EnvConfig.install_from_api?
+      json_formula = Homebrew::API::Formula.all_formulae[name]
+      json_formula = Homebrew::API.merge_variations(json_formula)
+      hsh["oldname"] = json_formula["oldname"]
+      hsh["requirements"] = json_formula["requirements"]
+    end
```

Comment:
Were these things missing even before the most recent regression?

Reply:
Yes. I hand tested by comparing with and without `HOMEBREW_NO_INSTALL_FROM_API`.

We do not currently read requirements or oldname from the JSON file. This will be fixed after 4.0.0.

Variations also doesn't work without the original source.